### PR TITLE
Spam filter

### PR DESCRIPTION
**Pull request by gtm-nayan**

### CODE REVIEW

File: src/events/on_message/_check_links.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	if (fails_link_test(message)) {
+		await delete_message(message);
+
+		await message.author
+			.send(
+				wrap_in_embed(
+					`Your message in ${message.channel} was removed since it doesn't contain a link, if you are trying to showcase a project please post a link with your text. Otherwise all conversation should be inside a thread\n\nYour message was sent below so you don't lose it!`,
+				),
+			)
+			.then((new_message) => new_message.reply(message.content));
```
Comment (ghostdevv) [inline src/events/on_message/_check_links.ts:15]: would this be better outside of the .then?

File: src/events/on_message/_common.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export function fails_link_test(message: Message) {
+	return in_link_only_channel(message) && !has_link(message);
+}
+
+export function in_link_only_channel(message: Message) {
+	return LINK_ONLY_CHANNELS.includes(message.channelId);
+}
+
+export const STOP = Symbol();
```
Comment (ghostdevv) [inline src/events/on_message/_common.ts:17]: What is the purpose of this?

**Review by ghostdevv — CHANGES_REQUESTED** [review]

File: src/events/on_message/_common.ts (hunk 2 above)
Comment (ghostdevv) [reply src/events/on_message/_common.ts:17]: Ah I see

File: src/events/on_message/_check_links.ts (hunk 1 above)
Comment (gtm-nayan) [reply src/events/on_message/_check_links.ts:15]: Would need an extra await and an const declaration. So maybe not.

Comment (ghostdevv) [reply src/events/on_message/_check_links.ts:15]: Not sure whether that is bad or not but lets keep how it is

**Review by ghostdevv — APPROVED** [review]

**Review by pngwn — APPROVED** [review]
Lgtm. We can always tweak if it is over eager.

**Pull request merged**